packages/blob/src/index.js

```diff
@@ -1,27 +1,41 @@
 /**
  * Browser dependencies
  */
-const { fetch } = window;
 const { createObjectURL, revokeObjectURL } = window.URL;
 
 const cache = {};
 
-export function createBlobURL( blob ) {
-	const url = createObjectURL( blob );
+/**
+ * Create a blob URL from a file.
+ *
+ * @param {File} file The file to create a blob URL for.
+ *
+ * @return {string} The blob URL.
+ */
+export function createBlobURL( file ) {
+	const url = createObjectURL( file );
 
-	cache[ url ] = blob;
+	cache[ url ] = file;
 
 	return url;
 }
 
+/**
+ * Retrieve a file based on a blob URL.
+ *
+ * @param {string} url The blob URL.
+ *
+ * @return {File} The file for the blob URL.
+ */
 export function getBlobByURL( url ) {
-	if ( cache[ url ] ) {
-		return Promise.resolve( cache[ url ] );
-	}
-
-	return fetch( url ).then( ( response ) => response.blob() );
+	return cache[ url ];
```

Comment:
Before we verified if the file exists in the cache, is it safe to assume that for all blob url's used here we used createBlobURL before? In that case, I think we should specify in the comments that this function only works for blubUrls created by createBlobURL.

Reply:
Sure, that hasn't changed. I'll add a comment.